qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,422 @@
+            circ = QuantumCircuit(self.num_qubits)
+            # Hack to get target bitstrings until qiskit-terra#9475 is resolved
+            circ_target = QuantumCircuit(self.num_qubits)
+            for layer in seq:
+                for elem in layer:
+                    circ.append(self._to_instruction(elem[1], basis_gates), elem[0])
+                    circ_target.append(self._to_instruction(elem[1]), elem[0])
+                circ.append(Barrier(self.num_qubits), circ.qubits)
+            circ.metadata = {
+                "xval": self.experiment_options.lengths[i % len(self.experiment_options.lengths)],
+                "group": "Clifford",
+                "physical_qubits": self.physical_qubits,
+                "target": compute_target_bitstring(circ_target),
+                "inverting_pauli_layer": self.experiment_options.inverting_pauli_layer,
+            }
+
+            if self.experiment_options.inverting_pauli_layer:
+                # Get target bitstring (ideal bitstring outputted by the circuit)
+                target = circ.metadata["target"]
+
+                # Pauli gates to apply to each qubit to reset each to the state 0.
+                # E.g., if the ideal bitstring is 01001, the Pauli label is IXIIX,
+                # which sets all qubits to 0 (up to a global phase)
+                label = "".join(["X" if char == "1" else "I" for char in target])
+                circ.append(Pauli(label), list(range(self._num_qubits)))
+
+            circ.measure_all()
+            circuits.append(circ)
+        return circuits
+
+    def _to_instruction(
+        self,
+        elem: SequenceElementType,
+        basis_gates: Optional[Tuple[str, ...]] = None,
+    ) -> Instruction:
+        """Convert the sampled object to an instruction."""
+        if isinstance(elem, Integral):
+            return _clifford_1q_int_to_instruction(elem, basis_gates)
+        elif isinstance(elem, Instruction):
+            return elem
+        elif getattr(elem, "to_instruction", None):
+            return elem.to_instruction()
+        else:
+            return elem()
+
+    def _inverse_layer(
+        self, layer: List[Tuple[Tuple[int, ...], GateTypeT]]
+    ) -> List[Tuple[Tuple[int, ...], GateTypeT]]:
+        """Generates the inverse layer of a Clifford mirror RB layer by inverting the
+        single-qubit Cliffords and keeping the two-qubit gate identical. See
+        :class:`.RBSampler` for the format of the layer.
+
+        Args:
+            layer: The input layer.
+
+        Returns:
+            The layer that performs the inverse operation to the input layer.
+
+        Raises:
+            QiskitError: If the layer has invalid format.
+        """
+        inverse_layer = []
+        for elem in layer:
+            if len(elem[0]) == 1 and np.issubdtype(type(elem[1]), int):
+                inverse_layer.append((elem[0], inverse_1q(elem[1])))
+            elif len(elem[0]) == 2 and elem[1] == CXGate:
```

Comment:
In 09fe450 I've added a `_self_adjoint_gates` list in `mirror_rb_experiment`—is there a better way to look up named self-adjoint gates in Qiskit? Otherwise we should keep this list updated.

The other potential issue is that this list includes gates like Swaps that don't generate n-qubit Cliffords, so the user should not interpret the result as the average infidelity of an n-qubit Clifford layer if they use all Swaps. The aim of the experiment is to let users choose custom layer distributions for their mirror circuits, so I think we should still allow users to specify any set of gates they want. I don't know if we should print a warning in this case, what do you think?

Reply:
I think it'd be better to allow users to specify any set of gets they want and I don't think we need to print a warning (I feels it's too much).